src/main/java/book/store/mapper/OrderMapper.java

```diff
@@ -0,0 +1,49 @@
+package book.store.mapper;
+
+import book.store.config.MapperConfig;
+import book.store.dto.order.OrderDto;
+import book.store.dto.order.PlaceOrderRequestDto;
+import book.store.model.CartItem;
+import book.store.model.Order;
+import book.store.model.OrderItem;
+import book.store.model.ShoppingCart;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.Named;
+
+@Mapper(config = MapperConfig.class, uses = {OrderItemMapper.class})
+public interface OrderMapper {
+    OrderDto toDto(Order order);
+
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "order", ignore = true)
+    @Mapping(target = "price", source = "book.price")
+    OrderItem cartItemToOrderItem(CartItem cartItem);
+
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "user", source = "shoppingCart.user")
+    @Mapping(target = "status", constant = "PENDING")
```

Comment:
Order already initialised with `status=Pending` and `orderDate = LocalDateTime.now()` so you can remove these mapping